
# INVESTMENT CASE STUDY  - DATA ANALYSIS                                                                
This is a group case study                                                                             
                                                                                                       
# Members:                                                                                               
 1. **Shadab Hussain**                                                                                      
 2. **Nidhi Tripathi**                                                                                 
 3. **Bhanu Pratap**                                                                                       
 4. **vikash kumar**                                                                               
                                                                                                    
# Brief on subject:                                                                                     
 Spark Funds is an asset management company. Spark Funds wants to make investments in a few companies. 
 Spark Funds wants to understand the global trends in investments so that they can take the investment  
 decisions effectively.                                                                                 
                                                                                                        
 **Business objective**: The objective is to identify the best sectors, countries, and a suitable investment type for making investments. The overall strategy is to invest where others are investing,implying that the best sectors and countries are the ones where most investments are happening. 
 
 
 
 

# Input files used: 
1. **Companies.txt**  - Provides details of companies involved for analysis. It provides details about the country , region which it belongs to. Permalink is the unique key for each company.    

|     Attribute              |  Description                                       |
|----------------------------|--------------------------------------------------- |
|   Permalink                |     Unique ID of company                           |
|   name                     |     Company name                                   |
|   homepage_url             |     Website URL                                    |
|   category_list            |     Category/categories to which a company belongs |
|   status                   |     Operational status                             |
|   country_code             |     Country Code                                   |
|   state_code               |     State                                          |                                                            


2. **Rounds2.csv** - Describes the details of funding happened for companies. funding_round_permalink is the unique ID of the funding round data.

|     Attribute                |  Description                                             |
|------------------------------|----------------------------------------------------------|
|        company_permalink     |     Unique ID of company                                 |
|  funding_round_permalink     |     Unique ID of funding round                           |
|       funding_round_type     |     Type of funding – venture, angel, private equity etc.|
|       funding_round_code     |     Round of venture funding (round A, B etc.)           |
|                funded_at     |     Date of funding                                      |
|        raised_amount_usd     |     Money raised in funding (USD)                        |

3. **mapping.csv** - This file maps the numerous category names in the companies table (such 3D printing, aerospace, agriculture, etc.) to eight broad sector names. The purpose is to simplify the analysis into eight sector buckets, rather than trying to analyse hundreds of them.

# Valid sectors available for mapping in Mapping.csv                                               
  1. Automotive & Sports                                                                                
  2. Cleantech / Semiconductors                                                                         
  3. Entertainment                                                                                      
  4. Health                                                                                             
  5. Manufacturing                                                                                      
  6. News, Search and Messaging                                                                        
  7. Others                                                                                             
  8. Social, Finance, Analytics, Advertising 

In [1]:
# Supress Warnings

import warnings
warnings.filterwarnings('ignore')

In [2]:
# Useful imports
import pandas as pd

In [3]:
#Loading the companies.txt file into companies dataframe
companies = pd.read_csv('companies.txt', encoding = 'latin-1', sep = '\t', parse_dates = ['founded_at'])
companies.permalink = companies.permalink.apply(lambda x: x.encode('utf-8').decode('ascii','ignore'))

In [4]:
#Loading the rounds2.csv file into rounds2 datafram
rounds2 = pd.read_csv('rounds2.csv', parse_dates = ['funded_at'], encoding = 'iso-8859-1')
rounds2.company_permalink = rounds2.company_permalink.apply(lambda x: x.encode('utf-8').decode('ascii','ignore'))

In [5]:
#viewing the companies data frame
companies.head()

permalink                    name  \
0                  /Organization/-Fame                   #fame   
1               /Organization/-Qounter                :Qounter   
2  /Organization/-The-One-Of-Them-Inc-  (THE) ONE of THEM,Inc.   
3                /Organization/0-6-Com                 0-6.com   
4       /Organization/004-Technologies        004 Technologies   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2                http://oneofthem.jp   
3                 http://www.0-6.com   
4  http://004gmbh.de/en/004-interact   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2                                  Apps|Games|Mobile  operating          NaN   
3                                        Curated Web  operating          CHN   
4                                           Software  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         Mumbai         nan  
1         DE             DE - Other  Delaware City  04-09-2014  
2        NaN                    NaN            NaN         nan  
3         22                Beijing        Beijing  01-01-2007  
4         IL  Springfield, Illinois      Champaign  01-01-2010

In [6]:
#viewing the rounds2 data frame
rounds2.head()

company_permalink  \
0                  /organization/-fame   
1               /ORGANIZATION/-QOUNTER   
2               /organization/-qounter   
3  /ORGANIZATION/-THE-ONE-OF-THEM-INC-   
4                /organization/0-6-com   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   

  funding_round_code  funded_at  raised_amount_usd  
0                  B 2015-05-01         10000000.0  
1                  A 2014-10-14                NaN  
2                NaN 2014-01-03           700000.0  
3                  B 2014-01-30          3406878.0  
4                  A 2008-03-19          2000000.0

In [7]:
#making a copy of both dataframe for future use
companies_copy = companies.copy()
rounds2_copy = rounds2.copy()


# Checkpoint 1: Data Cleaning 1

In [8]:
#Checking the structure of companies data frame
companies.shape

(66368, 10)

In [9]:
#listing the colunm and data type of companies dataframe
companies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 66368 entries, 0 to 66367
Data columns (total 10 columns):
permalink        66368 non-null object
name             66367 non-null object
homepage_url     61310 non-null object
category_list    63220 non-null object
status           66368 non-null object
country_code     59410 non-null object
state_code       57821 non-null object
region           58338 non-null object
city             58340 non-null object
founded_at       66368 non-null object
dtypes: object(10)
memory usage: 5.1+ MB


In [10]:
#Checking the structure of rounds2 data frame
rounds2.shape

(114949, 6)

In [11]:
#listing the colunm and data type of rounds2 dataframe
rounds2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114949 entries, 0 to 114948
Data columns (total 6 columns):
company_permalink          114949 non-null object
funding_round_permalink    114949 non-null object
funding_round_type         114949 non-null object
funding_round_code         31140 non-null object
funded_at                  114949 non-null datetime64[ns]
raised_amount_usd          94959 non-null float64
dtypes: datetime64[ns](1), float64(1), object(4)
memory usage: 5.3+ MB


In [12]:
#Converting the permalink column in companies to lower case
companies['permalink'] = companies['permalink'].str.lower()

In [13]:
#Converting the company_permalink column in rounds2 to lower case
rounds2['company_permalink'] = rounds2['company_permalink'].str.lower()

In [14]:
#How many unique companies are present in rounds2?
rounds2['company_permalink'].nunique()

66368

Number of Unique companies present in rounds2 are **66368**

In [15]:
#How many unique companies are present in companies?
companies['permalink'].nunique()

66368

Number of Unique companies present in companies are **66368**
Total number of rows in companies are **66368** and number of unique values of permalink are also **66368** so **permalink**
in companies data frame can be used as unique key for each company.


In [16]:
#Are there any companies in the rounds2 file which are not present in companies? 
len(rounds2.loc[~rounds2['company_permalink'].isin(companies['permalink']), :])

0

## NO
All the companies in the rounds2 file are present in companies files.

In [17]:
#Merge the two data frames so that all variables (columns) in the companies frame 
#are added to the rounds2 data frame.Name the merged frame master_frame.

master_frame = rounds2.merge(companies, how = 'inner', left_on = 'company_permalink', right_on = 'permalink')
master_frame.head(10)


company_permalink  \
0                  /organization/-fame   
1               /organization/-qounter   
2               /organization/-qounter   
3  /organization/-the-one-of-them-inc-   
4                /organization/0-6-com   
5       /organization/004-technologies   
6     /organization/01games-technology   
7  /organization/0ndine-biomedical-inc   
8  /organization/0ndine-biomedical-inc   
9                 /organization/0xdata   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
1  /funding-round/22dacff496eb7acb2b901dec1dfe5633            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
3  /funding-round/650b8f704416801069bb178a1418776b            venture   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
5  /funding-round/1278dd4e6a37fa4b7d7e06c21b3c1830            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   
8  /funding-round/954b9499724b946ad8c396a57a5f3b72            venture   
9  /funding-round/383a9bd2c04f7038bb543ccef5ba3eae               seed   

  funding_round_code  funded_at  raised_amount_usd  \
0                  B 2015-05-01         10000000.0   
1                  A 2014-10-14                NaN   
2                NaN 2014-01-03           700000.0   
3                  B 2014-01-30          3406878.0   
4                  A 2008-03-19          2000000.0   
5                NaN 2014-07-24                NaN   
6                NaN 2014-01-07            41250.0   
7                NaN 2009-11-09            43360.0   
8                NaN 2009-12-21           719491.0   
9                NaN 2013-05-22          3000000.0   

                             permalink                    name  \
0                  /organization/-fame                   #fame   
1               /organization/-qounter                :Qounter   
2               /organization/-qounter                :Qounter   
3  /organization/-the-one-of-them-inc-  (THE) ONE of THEM,Inc.   
4                /organization/0-6-com                 0-6.com   
5       /organization/004-technologies        004 Technologies   
6     /organization/01games-technology      01Games Technology   
7  /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   
8  /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   
9                 /organization/0xdata                  H2O.ai   

                        homepage_url  \
0                 http://livfame.com   
1             http://www.qounter.com   
2             http://www.qounter.com   
3                http://oneofthem.jp   
4                 http://www.0-6.com   
5  http://004gmbh.de/en/004-interact   
6             http://www.01games.hk/   
7               http://ondinebio.com   
8               http://ondinebio.com   
9                     http://h2o.ai/   

                                       category_list     status country_code  \
0                                              Media  operating          IND   
1  Application Platforms|Real Time|Social Network...  operating          USA   
2  Application Platforms|Real Time|Social Network...  operating          USA   
3                                  Apps|Games|Mobile  operating          NaN   
4                                        Curated Web  operating          CHN   
5                                           Software  operating          USA   
6                                              Games  operating          HKG   
7                                      Biotechnology  operating          CAN   
8                                      Biotechnology  operating          CAN   
9                                          Analytics  operating          USA   

  state_code                 region           city  founded_at  
0         16                 Mumbai         M

In [18]:
#making the copy of master_frame for future use
master_frame_copy = master_frame.copy()

In [19]:
#How many observations are present in master_frame ?
master_frame.shape[0]

114949

Number of observations present in master_frame are **114949**

# Data Cleaning

In [20]:
# Write your code for column-wise null count here
master_frame.isnull().sum()

company_permalink              0
funding_round_permalink        0
funding_round_type             0
funding_round_code         83809
funded_at                      0
raised_amount_usd          19990
permalink                      0
name                           1
homepage_url                6134
category_list               3410
status                         0
country_code                8678
state_code                 10946
region                     10167
city                       10164
founded_at                     0
dtype: int64

In [21]:
# Write your code for column-wise null percentages here
round((100*master_frame.isnull().sum()/len(master_frame.index)),2)

company_permalink           0.00
funding_round_permalink     0.00
funding_round_type          0.00
funding_round_code         72.91
funded_at                   0.00
raised_amount_usd          17.39
permalink                   0.00
name                        0.00
homepage_url                5.34
category_list               2.97
status                      0.00
country_code                7.55
state_code                  9.52
region                      8.84
city                        8.84
founded_at                  0.00
dtype: float64

In [22]:
# Dropping the rows where value for raised_amount_usd is not present.
master_frame = master_frame.dropna(axis = 0, subset = ['raised_amount_usd'])


In [23]:
## Dropping the rows where value for country_code is not present.
master_frame = master_frame.dropna(axis = 0, subset = ['country_code'])

In [24]:
# Dropping the rows where value for raised_amount_usd is 0. As this will give us wrong answer while taking the average of 
# of raised_amount_usd or ddoing any type of aggegation function.
master_frame = master_frame[~(master_frame['raised_amount_usd'] == 0)]

In [25]:
#Checking the shape of clean dataframe
master_frame.shape


(88729, 16)

In [26]:
#making the copy of clean master_frame for future use
master_frame_copy = master_frame.copy()


In [27]:
# Write your code for column-wise null percentages here
round((100*master_frame.isnull().sum()/len(master_frame.index)),2)

company_permalink           0.00
funding_round_permalink     0.00
funding_round_type          0.00
funding_round_code         69.66
funded_at                   0.00
raised_amount_usd           0.00
permalink                   0.00
name                        0.00
homepage_url                3.73
category_list               0.63
status                      0.00
country_code                0.00
state_code                  1.97
region                      1.35
city                        1.34
founded_at                  0.00
dtype: float64

Now we have 0.0% misssing value for `raised_amount_usd` and `country_code` column, and we can do our analysis on these coloumn.

# Checkpoint 2: Funding Type Analysis

In [28]:
# Filtering the master data frame for four funding types (venture, angel, seed, and private equity)
master_frame_FT = master_frame[master_frame['funding_round_type'].isin(['seed', 'angel', 'venture', 'private_equity'])]

In [29]:
master_frame_FT.shape


(75476, 16)

In [30]:
Funding_Types = master_frame_FT.groupby('funding_round_type').mean()

In [31]:
#Calculate the average investment amount for each of the four funding types (venture, angel, seed, and private equity)
Funding_Types['raised_amount_usd'] = Funding_Types['raised_amount_usd'].astype('float').apply(lambda x : '{:.2f}'.format(x))
Funding_Types

raised_amount_usd
funding_round_type                  
angel                      974503.35
private_equity           73658704.60
seed                       750756.35
venture                  11742369.78

In [32]:
#Considering that Spark Funds wants to invest between 5 to 15 million USD per investment round, which investment type is 
#the most suitable for it?
Funding_Types[Funding_Types['raised_amount_usd'].astype('float').between(5000000, 15000000)] 

raised_amount_usd
funding_round_type                  
venture                  11742369.78

# Checkpoint 3: Country Analysis

In [33]:
master_frame_venture = master_frame[master_frame['funding_round_type'] == 'venture']
master_frame_venture.shape

(48084, 16)

In [34]:
top9 = master_frame_venture.groupby('country_code').sum().sort_values('raised_amount_usd', ascending = False).head(9)
top9['raised_amount_usd'] = top9['raised_amount_usd'].apply(lambda x : '{:.2f}'.format(x))
top9


raised_amount_usd
country_code                  
USA            422510842796.00
CHN             39835418773.00
GBR             20245627416.00
IND             14391858718.00
CAN              9583332317.00
FRA              7259536732.00
ISR              6907514579.00
DEU              6346959822.00
JPN              3363676611.00

## Calculating the English Speeaking Countries List

I have calulated the country codes for the English speaking countries from the pdf given to us and i have created a list of all the English spaeking countries and saved those in the list called **English_speaking_countries**.

In [35]:

English_speaking_countries = ['ATG','AUS', 'BRB','BLZ','BWA','CMR','CAN','DMA','ERI','ETH','FSM','FJI','GHA','GRD','GUY','IND',
                               'IRL','JAM', 'KEN', 'KIR', 'LSO', 'LBR', 'MWI','MLT','MHL','MUS','NAM', 'NRU', 'NZL', 'NGA',
                              'PAK','PLW','PNG','PHL','RWA','KNA','LCA','VCT','WSM','SYC','SLE','SGP','SLB','ZAF','SSD',
                              'SDN','SWZ','TZA', 'BAH', 'GMB', 'TON', 'TTO', 'TUV', 'UGA', 'GBR', 'USA', 'VUT', 'ZMB', 'ZWE']

In [36]:
#Identify the top three English-speaking countries in the data frame top9
top3_English_speaking_countries = top9[top9.index.isin(English_speaking_countries)].head(3)
top3_English_speaking_countries

raised_amount_usd
country_code                  
USA            422510842796.00
GBR             20245627416.00
IND             14391858718.00

## Top 3 English Speaking Country 
1. USA (United Sates)
2. GBR (United Kingdom)
3. IND (India)

# Checkpoint 4: Sector Analysis 1

In [37]:
# reading the mapping file
mapping = pd.read_csv('mapping.csv', encoding = 'latin-1')
mapping.head()

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0            NaN                    0       1                           0   
1             3D                    0       0                           0   
2    3D Printing                    0       0                           0   
3  3D Technology                    0       0                           0   
4     Accounting                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0  
1                                        0  
2                                        0  
3                                        0  
4                                        1

## Observation from Mapping.csv file
1. category_list field contains '0' in strings where it is suppose to be 'na' <br>
  I have replaced the '0' with 'na' after this operation i noticed that there is a category_list "Enterprise2.0" 
  which now has been changed to "Enterprise2.na". So we have to convert it back to original form.
  Below code will do above 2 operation

In [38]:
# Replacing '0' with 'na' in the category_list column.
mapping['category_list'].replace(to_replace='0', value='na', regex = True, inplace = True)
mapping['category_list'].replace(to_replace='2.na', value='2.0', regex = True, inplace = True)
mapping.head()

category_list  Automotive & Sports  Blanks  Cleantech / Semiconductors  \
0            NaN                    0       1                           0   
1             3D                    0       0                           0   
2    3D Printing                    0       0                           0   
3  3D Technology                    0       0                           0   
4     Accounting                    0       0                           0   

   Entertainment  Health  Manufacturing  News, Search and Messaging  Others  \
0              0       0              0                           0       0   
1              0       0              1                           0       0   
2              0       0              1                           0       0   
3              0       0              1                           0       0   
4              0       0              0                           0       0   

   Social, Finance, Analytics, Advertising  
0                                        0  
1                                        0  
2                                        0  
3                                        0  
4                                        1

In [39]:
#copying the dataframe
master_frame_split_cat_list = master_frame
master_frame_split_cat_list.shape

(88729, 16)

In [40]:
# Extracting the first category from category_list column and adding the new column primary_sector with thi svalue.
master_frame_split_cat_list['primary_sector'] = master_frame_split_cat_list.category_list.str.split('|', expand = True)[0]
master_frame_split_cat_list.head()

company_permalink  \
0                  /organization/-fame   
2               /organization/-qounter   
4                /organization/0-6-com   
6     /organization/01games-technology   
7  /organization/0ndine-biomedical-inc   

                           funding_round_permalink funding_round_type  \
0  /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
2  /funding-round/b44fbb94153f6cdef13083530bb48030               seed   
4  /funding-round/5727accaeaa57461bd22a9bdd945382d            venture   
6  /funding-round/7d53696f2b4f607a2f2a8cbb83d01839        undisclosed   
7  /funding-round/2b9d3ac293d5cdccbecff5c8cb0f327d               seed   

  funding_round_code  funded_at  raised_amount_usd  \
0                  B 2015-05-01         10000000.0   
2                NaN 2014-01-03           700000.0   
4                  A 2008-03-19          2000000.0   
6                NaN 2014-01-07            41250.0   
7                NaN 2009-11-09            43360.0   

                             permalink                    name  \
0                  /organization/-fame                   #fame   
2               /organization/-qounter                :Qounter   
4                /organization/0-6-com                 0-6.com   
6     /organization/01games-technology      01Games Technology   
7  /organization/0ndine-biomedical-inc  Ondine Biomedical Inc.   

             homepage_url                                      category_list  \
0      http://livfame.com                                              Media   
2  http://www.qounter.com  Application Platforms|Real Time|Social Network...   
4      http://www.0-6.com                                        Curated Web   
6  http://www.01games.hk/                                              Games   
7    http://ondinebio.com                                      Biotechnology   

      status country_code state_code      region           city  founded_at  \
0  operating          IND         16      Mumbai         Mumbai         nan   
2  operating          USA         DE  DE - Other  Delaware City  04-09-2014   
4  operating          CHN         22     Beijing        Beijing  01-01-2007   
6  operating          HKG        NaN   Hong Kong      Hong Kong         nan   
7  operating          CAN         BC   Vancouver      Vancouver  01-01-1997   

          primary_sector  
0                  Media  
2  Application Platforms  
4            Curated Web  
6                  Games  
7          Biotechnology

In [41]:
master_frame_split_cat_list.shape

(88729, 17)

In [42]:
#creating a series for main sectors
main_sector  = pd.Series(['Automotive & Sports', 'Blanks', 'Cleantech / Semiconductors', 'Entertainment', 
                'Health', 'Manufacturing', 'News, Search and Messaging', 'Others', 'Social, Finance, Analytics, Advertising'])

In [43]:
cat_to_sector_mapping = pd.melt(mapping, id_vars = ['category_list'], var_name = 'main_sector', value_name = 'present')
cat_to_sector_mapping = cat_to_sector_mapping[cat_to_sector_mapping['present'] == 1]
cat_to_sector_mapping.drop(labels = 'present',axis = 1, inplace =  True)
cat_to_sector_mapping.head()

category_list          main_sector
8   Adventure Travel  Automotive & Sports
14         Aerospace  Automotive & Sports
45              Auto  Automotive & Sports
46   Automated Kiosk  Automotive & Sports
47        Automotive  Automotive & Sports

In [44]:
#merging the cat_to_sector_mapping dataframe to master_frame_split_cat_list to map the primary_sector with main_Sector
merge_split_mapped = master_frame_split_cat_list.merge(cat_to_sector_mapping, how = 'inner', left_on = 'primary_sector',
                                                       right_on = 'category_list')

In [45]:
#droping the extra column
merge_split_mapped.drop(columns = 'category_list_y', inplace = True)

In [46]:
# renaming the category_list_x to  original name category_list
merge_split_mapped.rename(columns={'category_list_x': 'category_list'}, inplace=True)


In [47]:
merge_split_mapped.shape

(88424, 18)

In [48]:
merge_split_mapped.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 88424 entries, 0 to 88423
Data columns (total 18 columns):
company_permalink          88424 non-null object
funding_round_permalink    88424 non-null object
funding_round_type         88424 non-null object
funding_round_code         26834 non-null object
funded_at                  88424 non-null datetime64[ns]
raised_amount_usd          88424 non-null float64
permalink                  88424 non-null object
name                       88423 non-null object
homepage_url               85130 non-null object
category_list              87861 non-null object
status                     88424 non-null object
country_code               88424 non-null object
state_code                 86686 non-null object
region                     87236 non-null object
city                       87239 non-null object
founded_at                 88424 non-null object
primary_sector             87861 non-null object
main_sector                88424 non-null object


# Checkpoint 5: Sector Analysis 2

In [49]:
#Creating the D1 data frame for USA.
D1 = merge_split_mapped[merge_split_mapped['country_code'].isin(['USA']) &
                        merge_split_mapped['funding_round_type'].isin(['venture'])  &
                        merge_split_mapped['raised_amount_usd'].astype('float').between(5000000, 15000000)]             
D1.shape

(12098, 18)

In [50]:
#Adding the 2 cilumns Total_number_investment and Total_amount_invested
D1['Total_number_investment'] = D1.groupby(['main_sector'])['main_sector'].transform('count')
D1['Total_amount_invested'] = D1.groupby('main_sector')['raised_amount_usd'].transform('sum')
D1['Total_amount_invested'] = D1['Total_amount_invested'].apply(lambda x : '{:.2f}'.format(x))
D1.head()

company_permalink  \
7   /organization/all-def-digital   
30       /organization/chefs-feed   
58   /organization/huffingtonpost   
59   /organization/huffingtonpost   
80        /organization/matchmine   

                            funding_round_permalink funding_round_type  \
7   /funding-round/452a2342fe720285c3b92e9bd927d9ba            venture   
30  /funding-round/adca195749ae9ace84684723fbe75e5b            venture   
58  /funding-round/7f05940c4d2dfecb8e50a0e5720e5065            venture   
59  /funding-round/9241ae16e08df17ebdc064e49e23035a            venture   
80  /funding-round/41ac526630da57ad6eb9d02431b17657            venture   

   funding_round_code  funded_at  raised_amount_usd  \
7                   A 2014-06-08          5000000.0   
30                  A 2015-02-26          5000000.0   
58                  A 2006-01-08          5000000.0   
59                  B 2007-01-09          5000000.0   
80                  A 2007-01-09         10000000.0   

                        permalink                 name  \
7   /organization/all-def-digital      All Def Digital   
30       /organization/chefs-feed            ChefsFeed   
58   /organization/huffingtonpost  The Huffington Post   
59   /organization/huffingtonpost  The Huffington Post   
80        /organization/matchmine            MatchMine   

                     homepage_url                           category_list  \
7        http://alldefdigital.com                                   Media   
30       http://www.chefsfeed.com     Media|Mobile|Restaurants|Technology   
58  http://www.huffingtonpost.com                   Media|News|Publishing   
59  http://www.huffingtonpost.com                   Media|News|Publishing   
80           http://matchmine.com  Media|News|Reviews and Recommendations   

       status country_code state_code         region           city  \
7   operating          USA         CA    Los Angeles    Los Angeles   
30  operating          USA         CA    SF Bay Area  San Francisco   
58   acquired          USA         NY  New York City       New York   
59   acquired          USA         NY  New York City       New York   
80     closed          USA         MA         Boston        Needham   

    founded_at primary_sector    main_sector  Total_number_investment  \
7          nan          Media  Entertainment                      591   
30  01-01-2012          Media  Entertainment                      591   
58  09-05-2005          Media  Entertainment                      591   
59  09-05-2005          Media  Entertainment                      591   
80  01-01-2007          Media  Entertainment                      591   

   Total_amount_invested  
7          5099197982.00  
30         5099197982.00  
58         5099197982.00  
59         5099197982.00  
80         5099197982.00

In [51]:
#Creating the D2 data frame for GBR.
D2 = merge_split_mapped[merge_split_mapped['country_code'].isin(['GBR']) &
                        merge_split_mapped['funding_round_type'].isin(['venture'])  &
                        merge_split_mapped['raised_amount_usd'].astype('float').between(5000000, 15000000)]             
D2.shape

(626, 18)

In [52]:
#Adding the 2 cilumns Total_number_investment and Total_amount_invested
D2['Total_number_investment'] = D2.groupby(['main_sector'])['main_sector'].transform('count')
D2['Total_amount_invested'] = D2.groupby('main_sector')['raised_amount_usd'].transform('sum')
D2['Total_amount_invested'] = D2['Total_amount_invested'].astype('float').apply(lambda x : '{:.2f}'.format(x))
D2.head()

company_permalink  \
1                          /organization/90min   
2                          /organization/90min   
218  /organization/common-interest-communities   
250                /organization/geospock-ltd-   
251                /organization/geospock-ltd-   

                             funding_round_permalink funding_round_type  \
1    /funding-round/21a2cbf6f2fb2a1c2a61e04bf930dfe6            venture   
2    /funding-round/bd626ed022f5c66574b1afe234f3c90d            venture   
218  /funding-round/8195587cbd5e51af7514ee92ef4ba6ba            venture   
250  /funding-round/cf3fe3b7c86186b9f478d0ea37613f7a            venture   
251  /funding-round/e5e4ef4ebae63fc36ef0cd57dd20ff1c            venture   

    funding_round_code  funded_at  raised_amount_usd  \
1                  NaN 2015-06-10         15000000.0   
2                  NaN 2013-07-05          5800000.0   
218                NaN 2014-09-07         10000000.0   
250                NaN 2014-01-10          5460000.0   
251                  A 2015-05-10          5400000.0   

                                     permalink                         name  \
1                          /organization/90min                        90min   
2                          /organization/90min                        90min   
218  /organization/common-interest-communities  Common Interest Communities   
250                /organization/geospock-ltd-                GeoSpock Ltd.   
251                /organization/geospock-ltd-                GeoSpock Ltd.   

                              homepage_url  \
1                     http://www.90min.com   
2                     http://www.90min.com   
218  http://commoninterestcommunities.com/   
250                http://www.geospock.com   
251                http://www.geospock.com   

                                        category_list     status country_code  \
1                 Media|News|Publishing|Soccer|Sports  operating          GBR   
2                 Media|News|Publishing|Soccer|Sports  operating          GBR   
218  Application Platforms|Internet|Software|Startups  operating          GBR   
250         Application Platforms|Databases|Real Time  operating          GBR   
251         Application Platforms|Databases|Real Time  operating          GBR   

    state_code  region       city  founded_at         primary_sector  \
1           H9  London     London  01-01-2011                  Media   
2           H9  London     London  01-01-2011                  Media   
218         H9  London     London         nan  Application Platforms   
250         C3  London  Cambridge  01-01-2013  Application Platforms   
251         C3  London  Cambridge  01-01-2013  Application Platforms   

                    main_sector  Total_number_investment Total_amount_invested  
1                 Entertainment                       56          482784687.00  
2                 Entertainment                       56          482784687.00  
218  News, Search and Messaging                       73          615746235.00  
250  News, Search and Messaging                       73          615746235.00  
251  News, Search and Messaging                       73          615746235.00

In [53]:
#Creating the D3 data frame for IND.
D3 = merge_split_mapped[merge_split_mapped['country_code'].isin(['IND']) &
                        merge_split_mapped['funding_round_type'].isin(['venture'])  &
                        merge_split_mapped['raised_amount_usd'].astype('float').between(5000000, 15000000)]             
D3.shape

(330, 18)

In [54]:
#Adding the 2 cilumns Total_number_investment and Total_amount_invested
D3['Total_number_investment'] = D3.groupby(['main_sector'])['main_sector'].transform('count')
D3['Total_amount_invested'] = D3.groupby('main_sector')['raised_amount_usd'].transform('sum')
D3['Total_amount_invested'] = D3['Total_amount_invested'].astype('float').apply(lambda x : '{:.2f}'.format(x))
D3.head()

company_permalink  \
0               /organization/-fame   
542           /organization/babajob   
585  /organization/bharat-matrimony   
632     /organization/bluestone-com   
634     /organization/bluestone-com   

                             funding_round_permalink funding_round_type  \
0    /funding-round/9a01d05418af9f794eebff7ace91f638            venture   
542  /funding-round/b72eaac5ea12ac0f50573ac3d6d46b8d            venture   
585  /funding-round/e37673bc7b0f1dfd3782f8f7abdb9ec8            venture   
632  /funding-round/452a7fc1f34df2d3dcda4e28234bc671            venture   
634  /funding-round/f5b252d6442ce231bb01586ca1821f63            venture   

    funding_round_code  funded_at  raised_amount_usd  \
0                    B 2015-05-01         10000000.0   
542                  B 2015-04-28         10000000.0   
585                  B 2008-05-02         11750000.0   
632                  A 2012-01-24          5000000.0   
634                  B 2014-03-18         10000000.0   

                          permalink              name  \
0               /organization/-fame             #fame   
542           /organization/babajob           Babajob   
585  /organization/bharat-matrimony  Bharat Matrimony   
632     /organization/bluestone-com     Bluestone.com   
634     /organization/bluestone-com     Bluestone.com   

                       homepage_url  \
0                http://livfame.com   
542          http://www.babajob.com   
585  http://www.bharatmatrimony.com   
632            http://bluestone.com   
634            http://bluestone.com   

                                         category_list     status  \
0                                                Media  operating   
542  Curated Web|Information Technology|Services|St...  operating   
585                           Curated Web|Match-Making  operating   
632                                        Curated Web  operating   
634                                        Curated Web  operating   

    country_code state_code     region       city  founded_at primary_sector  \
0            IND         16     Mumbai     Mumbai         nan          Media   
542          IND         19  Bangalore  Bangalore  27-08-2007    Curated Web   
585          IND         25    Chennai    Chennai  12-03-1969    Curated Web   
632          IND         19  Bangalore  Bangalore  01-01-2011    Curated Web   
634          IND         19  Bangalore  Bangalore  01-01-2011    Curated Web   

                    main_sector  Total_number_investment Total_amount_invested  
0                 Entertainment                       33          280830000.00  
542  News, Search and Messaging                       52          433834545.00  
585  News, Search and Messaging                       52          433834545.00  
632  News, Search and Messaging                       52          433834545.00  
634  News, Search and Messaging                       52          433834545.00

## Analysis for D1 

In [55]:
#1. Total number of investments (count)
D1['raised_amount_usd'].count()

12098

In [56]:
#2. Total amount of investment (USD)
D1['raised_amount_usd'].sum()

108083057956.0

In [57]:
#Top 3 sector (based on count of investments) and number of investment in them.
D1_grp = D1.groupby('main_sector')
D1_grp.agg({'Total_number_investment': 'max'}).sort_values('Total_number_investment', ascending = False).head(3)

Total_number_investment
main_sector                                                     
Others                                                      2950
Social, Finance, Analytics, Advertising                     2714
Cleantech / Semiconductors                                  2300

In [100]:
#For the top sector count-wise (point 3), which company received the highest investment?
#For point 4 (second best sector count-wise), which company received the highest investment?
D1_1 = pd.DataFrame(pd.pivot_table(D1, index = ['main_sector', 'name'], values = ['raised_amount_usd'],
                                     aggfunc = 'size').sort_values(ascending = False))
D1_1.head(20)

0
main_sector                             name                              
News, Search and Messaging              Airspan Networks                 9
Cleantech / Semiconductors              Tigo Energy                      8
                                        Biodesix                         8
Social, Finance, Analytics, Advertising NetBase Solutions                7
                                        SST Inc. (Formerly ShotSpotter)  7
Health                                  EndoGastric Solutions            7
Cleantech / Semiconductors              ForSight Labs                    7
Entertainment                           Avnera                           7
News, Search and Messaging              VeriSilicon Holdings             6
Health                                  PatientKeeper                    6
Automotive & Sports                     Zero Motorcycles Inc.            6
Others                                  Virtustream                      6
Social, Finance, Analytics, Advertising ChoiceStream                     6
                                        InstaMed                         6
News, Search and Messaging              GlassHouse Technologies          6
Cleantech / Semiconductors              Metabolon                        6
Social, Finance, Analytics, Advertising FirstRain                        6
Others                                  Capella Photonics                6
Cleantech / Semiconductors              SCIenergy                        6
Others                                  Black Duck Software              6

In main_sector **Others** company **Virtustream** received the highest investment.<br>
In main_sector **Social, Finance, Analytics, Advertising** company **SST Inc. (Formerly ShotSpotter)** received the highest                                                                                                                       investment.

## Analysis for D2

In [62]:
#1. Total number of investments (count)
D2['raised_amount_usd'].count()

626

In [63]:
#2. Total amount of investment (USD)
D2['raised_amount_usd'].sum()

5422993148.0

In [64]:
#Top 3 sector (based on count of investments) and number of investment in them.
D2_grp = D2.groupby('main_sector')
D2_grp.agg({'Total_number_investment': 'max'}).sort_values('Total_number_investment', ascending = False).head(3)

Total_number_investment
main_sector                                                     
Others                                                       147
Social, Finance, Analytics, Advertising                      133
Cleantech / Semiconductors                                   128

In [101]:
#For the top sector count-wise (point 3), which company received the highest investment?
#For point 4 (second best sector count-wise), which company received the highest investment?
D2_2 = pd.DataFrame(pd.pivot_table(D2, index = ['main_sector', 'name'], values = ['raised_amount_usd'],
                                     aggfunc = 'size').sort_values(ascending = False))
D2_2.head(20)

0
main_sector                             name                           
Automotive & Sports                     Autoquake                     4
News, Search and Messaging              Ubiquisys                     4
Others                                  SenSage                       4
                                        Electric Cloud                4
Social, Finance, Analytics, Advertising Celltick Technologies         4
Automotive & Sports                     GreenRoad Technologies        4
Social, Finance, Analytics, Advertising myThings                      4
Manufacturing                           Antenova                      3
Others                                  SilverRail Technologies       3
                                        Enigmatec                     3
News, Search and Messaging              mySupermarket                 3
                                        BaseKit                       3
                                        brylle                        3
Others                                  OpenCloud                     3
Social, Finance, Analytics, Advertising Garlik                        3
                                        Victor                        3
                                        VisualDNA                     3
Cleantech / Semiconductors              Horizon Discovery             3
Manufacturing                           Nujira                        3
Cleantech / Semiconductors              F-star Biotechnology Limited  3

In main_sector **Others** company **SenSage** received the highest investment.<br>
In main_sector **Social, Finance, Analytics, Advertising** company **Celltick Technologies** received the highest                                                                                                                       investment.

## Analysis for D3

In [67]:

#1. Total number of investments (count)
D3['raised_amount_usd'].count()

330

In [68]:
#2. Total amount of investment (USD)
D3['raised_amount_usd'].sum()

2976543602.0

In [69]:
#Top 3 sector (based on count of investments) and number of investment in them.
D3_grp = D3.groupby('main_sector')
D3_grp.agg({'Total_number_investment': 'max'}).sort_values('Total_number_investment', ascending = False).head(3)

Total_number_investment
main_sector                                                     
Others                                                       110
Social, Finance, Analytics, Advertising                       60
News, Search and Messaging                                    52

In [102]:
#For the top sector count-wise (point 3), which company received the highest investment?
#For point 4 (second best sector count-wise), which company received the highest investment?
D3_3 = pd.DataFrame(pd.pivot_table(D3, index = ['main_sector', 'name'], values = ['raised_amount_usd'],
                                     aggfunc = 'size').sort_values(ascending = False))
D3_3.head(20)

0
main_sector                             name                       
Cleantech / Semiconductors              Azure Power               4
Others                                  Myntra                    4
Social, Finance, Analytics, Advertising Manthan Systems           4
Others                                  FirstCry.com              3
                                        CommonFloor               3
                                        Pepperfry.com             3
Social, Finance, Analytics, Advertising Komli Media               3
News, Search and Messaging              GupShup                   3
Cleantech / Semiconductors              Perfint Healthcare        3
Others                                  Fashion & You             2
News, Search and Messaging              Bluestone.com             2
Others                                  iProf Learning Solutions  2
Health                                  MyDentist                 2
Entertainment                           Games2Win                 2
Others                                  Comat Technologies        2
Entertainment                           Collectabillia.com        2
                                        Cleartrip                 2
News, Search and Messaging              Guruji                    2
Social, Finance, Analytics, Advertising BankBazaar.com            2
Entertainment                           Yatra                     2

In main_sector **Others** company **Myntra** received the highest investment.<br>
In main_sector **Social, Finance, Analytics, Advertising** company **Manthan Systems** received the highest                                                                                                                       investment.